querybook/webapp/components/TranspileQueryModal/TranspileQueryModal.tsx

```diff
@@ -0,0 +1,149 @@
+import { diffWordsWithSpace } from 'diff';
+import React, { useCallback, useMemo } from 'react';
+
+import { IQueryEngine } from 'const/queryEngine';
+import { useResource } from 'hooks/useResource';
+import { TemplatedQueryResource } from 'resource/queryExecution';
+import { Button } from 'ui/Button/Button';
+import { CodeHighlightWithMark } from 'ui/CodeHighlight/CodeHighlightWithMark';
+import { IHighlightRange } from 'ui/CodeHighlight/types';
+import { Loading } from 'ui/Loading/Loading';
+import { Modal } from 'ui/Modal/Modal';
+import { AccentText } from 'ui/StyledText/StyledText';
+import { Tag } from 'ui/Tag/Tag';
+
+import './TranspileQueryModal.scss';
+
+interface IProps {
+    query: string;
+    transpilerName: string;
+    toEngine: IQueryEngine;
+    fromEngine: IQueryEngine;
+
+    onTranspileConfirm: (query: string, engine: IQueryEngine) => void;
+    onHide: () => void;
+}
+
+export const TranspileQueryModal: React.FC<IProps> = ({
+    query,
+    transpilerName,
+    toEngine,
+    fromEngine,
+    onHide,
+    onTranspileConfirm,
+}) => {
+    const { data: transpiledQuery, isLoading } = useResource(
+        useCallback(
+            () =>
+                TemplatedQueryResource.transpileQuery(
```

Comment:
previously it was called with `await TemplatedQueryResource.transpileQuery()` directly, what's the difference and benefit to use `useResource`

Reply:
useResource is just a hook to auto load the data